### PR TITLE
Added jsonp support

### PR DESCRIPTION
**Pull request by rob-mccann**
Hey,

I hope I did it right this time!

Here's support for JSONP in Controller_Rest and Format

The mime type is set to text/javascript because it's what jquery seems to like, but I'll leave that to your discretion.

Rob

### CODE REVIEW

**Comment by philsturgeon** [comment]
You closed it? Looked ok to me.

**Comment by rob-mccann** [comment]
I shouldn't be allowed to do things when I'm stressed.

I closed it because of the default $callback being response. For some reason I thought it should be 'callback'.

I reopened it.

File: classes/format.php [hunk 1]
```diff
@@ -224,6 +224,20 @@ class Format {
...
 
 	/**
+	 * To JSONP conversion
+	 *
+	 * @param mixed $data
+	 * @return string
+	 */ 
+	public function to_jsonp($data = null)
+	{
+		 $callback = \Input::get_post('callback', null));
```
Comment (philsturgeon) [inline classes/format.php:234]: You don't need the null in there, it does that already :)

**Comment by philsturgeon** [comment]
This looks great but it is out of date and I cannot merge it. Can you update your branch please?

**Comment by rob-mccann** [comment]
I think I've updated my branch. Let me know if I did it wrong!

**Comment by philsturgeon** [comment]
I'm still getting "This pull request cannot be automatically merged. ", which normally means random conflicts and I don't want to have to do that :p

**Comment by philsturgeon** [comment]
OH WE DONT HAVE DEVELOP ANYMORE!

Can you reapply this commit to 1.1/develop instead of develop?

**Comment by rob-mccann** [comment]
Are you able to merge my develop to your 1.1/develop? Otherwise, do I create a new pull request? (sorry, not done this before - still learning :) )

**Comment by philsturgeon** [comment]
Not a problem. Just 

git checkout -t origin/1.1/develop
git log develop
git cherry-pick <hash of whatever it was in the log>

Push that and send a new pull request. I COULD do it manually, but it's a learning process for everyone right? :D

**Comment by rob-mccann** [comment]
'teach a man to fish' and all that.

For reference: I created the new pull request [#562](https://github.com/fuel/core/pull/562).

**Pull request closed without merging**